### ISSUE TITLE
[Regression] `dbt deps` fails when there is no write access to the `package-lock.yml` file in the project root path

### ISSUE THREAD

**Issue by perttus** [dbt-labs/dbt-core#9415]
### Is this a regression in a recent version of dbt-core?

- [X] I believe this is a regression in dbt-core functionality
- [X] I have searched the existing issues, and I could not find an existing issue for this regression

### Current Behavior

`dbt deps` fails when there is no write access to the `package-lock.yml` file in the project root path. 

There is no option to skip the package-lock file. It doesn't help even if the file already exists. It doesn't help if I ignore the error since any other following run will fail because the packages are not installed correctly.
I'm using a `/tmp` folder for the `packages-install-path` where Airflow has write access. I'm using gitSync which doesn't allow write access to the project folder.

### Expected/Previous Behavior

Use `packages-install-path` instead for the lock file or add option to skip the lock file creation and checking.

### Steps To Reproduce

`chmod u-w .`
`dbt deps`

### Relevant log output

```shell
> OSError: [Errno 30] Read-only file system: '/opt/airflow/.../dags/<dbt_project_folder>/package-lock.yml'
```

### Environment

```markdown
- OS: apache/airflow:2.5.1-python3.8 (debian/kubernetes with gitsync)
- Python: 3.8
- dbt (working version): 1.6.0
- dbt (regression version): 1.7.3
```

### Which database adapter are you using with dbt?

other (mention it in "Additional Context")

### Additional Context

using Vertica adapter

**Comment by dbeatty10** [comment]
Thanks for reporting this @perttus !

> It doesn't help even if the file already exists

You tried generating the `package-lock.yml` via `dbt deps` in a writable environment and committed that file to your git repository?

**Comment by perttus** [comment]
> Thanks for reporting this @perttus !
> 
> > It doesn't help even if the file already exists
> 
> You tried generating the `package-lock.yml` via `dbt deps` in a writable environment and committed that file to your git repository?

Yes indeed. So is it supposed to work with that approach?

**Comment by dbeatty10** [comment]
> So is it supposed to work with that approach?

It worked for me, so I don't know what the key difference is between your environment and mine. Could you take a look at my commands and project below and try to determine what the difference is?

<details>
<summary>

### Project files
</summary>

My project setup doesn't need any models, just `dbt_project.yml`, `packages.yml`, and some environment variables. 

Here's my folder structure:
```
.
├── my_dbt_packages
├── my_dbt_project
│   ├── dbt_project.yml
│   ├── packages.yml
│   └── profiles.yml
├── my_logs
└── my_target
```

Include this within `dbt_project.yml`:

```yaml
packages-install-path: ../my_dbt_packages
```

`packages.yml`

```yaml
packages:
  - package: dbt-labs/dbt_utils
    version: 1.1.1
```

Here are my environment variables:

```
export DBT_LOG_PATH=/full/path/to/my_logs
export DBT_TARGET_PATH=/full/path/to/my_target
export DBT_PROJECT_DIR=my_dbt_project
export DBT_PROFILES_DIR=my_dbt_project
```

</details>

### Commands

This works for me:
```shell
chmod u+w my_dbt_project
dbt deps
```

And this works for me too (but it sounds like it does **not** work for you?):
```shell
chmod u-w my_dbt_project
dbt deps
```

But similar to what you reported, this doesn't work for me:
```shell
sudo rm my_dbt_project/package-lock.yml
dbt deps
```

But this works again:
```shell
chmod u+w my_dbt_project
dbt deps
```

**Comment by perttus** [comment]
Thank you for the help! I managed to resolve the issue by using exactly the same environment variables locally than in my Airflow environment. The problem was caused by different git url in `packages.yml` file because locally I'm using SSH key access and in Airflow environment I must use service account token instead. Turns out that the `sha1_hash` is different once the git url is resolved from the environment variables.

``` yml
packages:
  - git: "{{env_var('DBT_ENV_UTILS_GIT_URL', 'git@github.com:<org>/<repo>.git')}}"
    revision: 0.0.1
  - package: dbt-labs/dbt_utils
    version: 1.1.1
  - package: calogica/dbt_expectations
    version: 0.10.1
```

 I just needed to define the environment variable locally before running the `dbt deps` and commit the lock file afterwards. Now everything works again 🎉
